sdk-cpp/src/internal/do_download_property_internal.h

```diff
@@ -0,0 +1,75 @@
+#ifndef _DELIVERY_OPTIMIZATION_DO_DOWNLOAD_PROPERTY_INTERNAL_H
+#define _DELIVERY_OPTIMIZATION_DO_DOWNLOAD_PROPERTY_INTERNAL_H
+
+#if defined(DO_INTERFACE_COM)
+#include <OAIdl.h>
+#else
+#include <boost/variant.hpp>
+#endif
+
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "do_download_property.h"
+
+namespace microsoft
+{
+namespace deliveryoptimization
+{
+namespace details
+{
+class CDownloadPropertyValueInternal
```

Comment:
>CDownloadPropertyValueInternal [](http://example.com/codeflow?start=6&length=30)

We don't need a separate internal class when the native_type is controlled by a compile-time constant. Or am I missing something?

Reply:
Isn't the using directive already a compile time constant? (I may be misunderstanding here). But yes the reason for this is to abstract away the native_value type so that the header contains no traces of
(DO_INTERFACE_COM) or (DO_INTERFACE_REST)